src/components/Contacts/ContactDetails/ContactDetailContext.tsx

```diff
@@ -109,39 +109,59 @@ export const ContactDetailProvider: React.FC<Props> = ({ children }) => {
     setEditPersonModalOpen(undefined);
   }, [router]);
 
+  const contextValue = useMemo(
+    () => ({
+      editingAddressId,
+      setEditingAddressId,
+      addAddressModalOpen,
+      setAddAddressModalOpen,
+      editModalOpen,
+      setEditModalOpen,
+      editOtherModalOpen,
+      setEditOtherModalOpen,
+      editMailingModalOpen,
+      setEditMailingModalOpen,
+      selectedTabKey,
+      setSelectedTabKey,
+      handleTabChange,
+      editPersonModalOpen,
+      openPersonModal,
+      closePersonModal,
+      createPersonModalOpen,
+      setCreatePersonModalOpen,
+      selectedDonationTabKey,
+      setSelectedDonationTabKey,
+      notes,
+      setNotes,
+      referralsModalOpen,
+      setReferralsModalOpen,
+      deleteModalOpen,
+      setDeleteModalOpen,
+      anchorEl,
+      setAnchorEl,
+    }),
+    [
+      editingAddressId,
+      addAddressModalOpen,
+      editModalOpen,
+      editOtherModalOpen,
+      editMailingModalOpen,
+      selectedTabKey,
+      handleTabChange,
```

Comment:
You need to wrap `handleTabChange` in `useCallback` so that it will be stable across renders.